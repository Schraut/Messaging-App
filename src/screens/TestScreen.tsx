import React from "react";
import { View, Text } from "react-native";
import TestComponent from "../components/TestComponent";

const TestScreen = () => {
  return (
    <View>
      <Text>Test Screen</Text>
      <TestComponent />
    </View>
  );
};

export default TestScreen;
