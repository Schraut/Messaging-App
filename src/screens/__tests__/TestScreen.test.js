import React from "react";
import renderer from "react-test-renderer";
import TestScreen from "../test_screen/TestScreen";
import ThemeContent from "../../../resources/theme/theme";

describe("Default TestScreen", () => {
  it("renders in light mode", () => {
    const tree = renderer
      .create(
        <ThemeContent>
          <TestScreen />
        </ThemeContent>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    const tree = renderer
      .create(
        <ThemeContent isDarkMode={true}>
          <TestScreen />
        </ThemeContent>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
