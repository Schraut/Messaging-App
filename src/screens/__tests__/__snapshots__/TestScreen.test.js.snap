// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default TestScreen renders in dark mode 1`] = `
<View>
  <Text>
    Test Screen
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "height": 160,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 10,
          "fontWeight": "400",
        }
      }
    >
      Hello
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "green",
          "height": 20,
          "opacity": 1,
          "width": 80,
        }
      }
    />
  </View>
</View>
`;

exports[`Default TestScreen renders in light mode 1`] = `
<View>
  <Text>
    Test Screen
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "height": 160,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "System",
          "fontSize": 10,
          "fontWeight": "400",
        }
      }
    >
      Hello
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "green",
          "height": 20,
          "opacity": 1,
          "width": 80,
        }
      }
    />
  </View>
</View>
`;
